### PR TITLE
release/0.12.0

### PR DESCRIPTION
**Pull request by vanbasten17**
_Remember to tag the PR with **WIP** tag if it's not ready to be merged_.  
[Reference](https://blog.codeminer42.com/on-writing-a-great-pull-request-37c60ce6f31d/)

## Description
* Added changelog with all notable changes.
* Bump and publish all package versions to `0.12.0-rc.0` (except plugin-contenful and plugin-dynamo-db)  
* Move READMEs for every package from https://github.com/hubtype/botonic/pull/741 to this PR before publishing.

### Process

| Template           | Items verified | 
|----------------|---------------|
| blank | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| childs | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| dynamic-carousel | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| handoff | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| intent | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| custom-webchat | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| nlu | <ul><li>- [x] clean install</li><li>[X] botonic train: `SyntaxError: Invalid regular expression: /?/: Nothing to repeat`, published fix version `@botonic/nlu@0.12.0-rc.1`</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| tutorial | <ul><li>- [x] clean install</li><li>- [X] first deploy success</li><li>- [X] working in prod env</li></ul>  | 
| general | <ul><li>- [X] markdown working</li><li>- [X] links auto-detected</li><li>- [X] images with urls accepted</li><li>- [X] Botonic Logo not appearing for few milliseconds if another image is set (prod)</li><li>- [X] Scrollbar is set to bottom when webchat is opened</li><li>- [X] Default behavior for replies/buttons once clicked is to show the button's text</li><li>- [X] handoff working well in serve/prod envs</li><li>- [X] working webviews</li></ul>  | 
## Context
Releasing `0.12.0` soon.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+
+  - Multichannel:
+
+    - Customize `Multichannel` visualization.
+    - `Multichannel` to support letter indexes.
+    - Compact `Multichannel` `Text` and `Carousel`.
+    - `Multichannel` to allow separating messages with custom string.
+
+- [@botonic/plugin-contentful](https://www.npmjs.com/package/@botonic/plugin-contentful)
+  - `Comment` and `Chitchat` typing improvements.
```
Comment (dpinol) [inline CHANGELOG.md:63]: @vanbasten17 Not important. You can remove this line

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+  - Allow passing `Normalizer` to `CMS`.
+  - Added Markup (WhatsApp Support) and Markdown parsing.
```
Comment (dpinol) [inline CHANGELOG.md:65]: - Added markup support for WhatsApp and Markdown.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+  - Added stemmers/tokens for Russian.
+
```
Comment (dpinol) [inline CHANGELOG.md:67]: - Now normalizer throws an exception for empty texts.
- Keyword search now sets the result's score field

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+- [Project](https://github.com/hubtype/botonic)
+
+  - `pre-commit` was not aborting when `lint` failed
+  - Fix several `eslint` warnings.
+
+* [@botonic/cli](https://www.npmjs.com/package/@botonic/cli)
+  - Preserve `src/nlu` directories for all templates (added `.gitkeep` file). Remove them before running `botonic train`.
+  - Add missing calls with `await`.
+  - Use `rimraf` in favor of `fs.rmdirSync` which produced buggy behaviors.
```
Comment (dpinol) [inline CHANGELOG.md:100]: In release notes, I wouldn't explain how you solved the issues, but which problem you fixed

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+  - Better scrollbar handling when Webchat is hovered.
+  - Limit attachments sizes to 10MB.
+  - Fix attachments crashing in production build for Webchat (media messages will be no longer stored as binary data in local storage, so the content will only be available temporarily in `botonic serve` mode).
+  - Now the Webchat's theme properties `message.bot.image`, `header.image`, `intro.image` and `triggerButton.image` accept URLs.
+  - Fix issue causing Botonic Logo to be visible during few milliseconds in production.
+  - Handle parent page scrolling issues in Webchat for iOS.
+
+* [@botonic/plugin-contentful](https://www.npmjs.com/package/@botonic/plugin-contentful)
+  - `BotonicMsgConverter` was not using `replaceEmptyStringsWith` option with value.
+  - Recursively transform `MessageContents`.
```
Comment (dpinol) [inline CHANGELOG.md:117]: Remove: 
  " - Recursively transform `MessageContents`."

Add:
- New operations to clone Contents performing composable transformations.
- New operations which traverse the chain of followUp fields of MessageContent's

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+  - Prefer tokens to stems when comparing to keywords.
```
Comment (dpinol) [inline CHANGELOG.md:118]: When calculating the match substring and score between an utterance and a keyword, prefer tokens to stems when comparing to keywords.

File: CHANGELOG.md [hunk 7]
```diff
@@ -11,6 +11,112 @@ All notable changes to Botonic will be documented in this file.
...
+  - Use `rimraf` in favor of `fs.rmdirSync` which produced buggy behaviors.
+  - Templates: Call `CleanWebpackPlugin` with  
+    `{ cleanOnceBeforeBuildPatterns: ['dist'] }` to prevent occasional builds from crashing. We strongly suggest to add this line in your bot's `webpack.config.js` if updating from previous versions.
+
+- [@botonic/react](https://www.npmjs.com/package/@botonic/react)
+
+  - Fixed sending wrong payloads when a handover was ended in `botonic serve` mode.
+  - `postMessage` promises were not forwarded.
+  - Better scrollbar handling when Webchat is hovered.
+  - Limit attachments sizes to 10MB.
```
Comment (dpinol) [inline CHANGELOG.md:109]: attachments sizes => attachment sizes

**Comment by vanbasten17** [comment]
Merging, all packages published.

**Pull request merged**